src/get-binary-path.ts

```diff
@@ -0,0 +1,48 @@
+import * as path from "path";
+import * as vscode from "vscode";
+import { existsSync } from "fs";
+
+const getWorkspaceFolderFsPath = () => {
+  if (vscode.workspace.workspaceFolders === undefined) {
+    console.log("workspace folders was undefined");
+    return;
+  }
+  if (vscode.workspace.workspaceFolders.length === 0) {
+    console.log("workspace folders was not set");
+    return;
+  }
+  const uri = vscode.workspace.workspaceFolders[0].uri;
+  if (uri.scheme !== "file") {
+    console.log("uri was not file: ", uri.scheme);
+    return;
+  }
+  return uri.fsPath;
+};
+
+export const getBinaryPath = () => {
+  const workspaceFolderFsPath = getWorkspaceFolderFsPath();
+  if (workspaceFolderFsPath === undefined) {
+    return {};
+  }
+  let binaryPath = vscode.workspace
+    .getConfiguration("buf")!
+    .get<string>("binaryPath");
+  if (binaryPath === undefined) {
+    console.log("buf binary path was not set");
+    return {};
+  }
+  const relative = binaryPath.startsWith(".");
```

Comment:
Is there validation that every relative path we are taking here starts with the `.`/`..` notation? To me, `node_modules/.bin/buf` would be a valid relative path, for example. Would it make more sense to check `path.isAbsolute` here so that this is OS agnostic?

Reply:
it was originally `path.isAbsolute` but that also identified the default `buf` as relative, which isn't true. The confusing part is that the default setting really isn't a path, it's just the exec command.

Reply:
Ah, I see, it's because it's not exactly a "binary path"

https://github.com/bufbuild/vscode-buf/blob/c256ac203f4b402ed6d4bdb2e0d34af9bccb83df/package.json#L60-L64

Thoughts on the following approach?
- Check for exactly the default value, "buf" -- if this works, it's simply calling out to `buf` in your shell, and that defaults to whatever is in your `$PATH`
- If this value is not the default value, then we check `path.isAbsolute`

I think the alternate approach is to change the default to actually use `$PATH`, but this feels quite fragile/not really OS-agnostic.